### PR TITLE
Clean up the JIT's histogram type.

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

**Comment by pgavlin** [comment]
@dotnet/jit-contrib PTAL

File: src/jit/utils.cpp [hunk 1]
```diff
@@ -778,109 +778,92 @@ void ConfigMethodRange::initRanges(__in_z LPCWSTR rangeStr)
...
 }
 
-histo::~histo()
+Histogram::~Histogram()
 {
-    histoAlloc->Free(histoCounts);
+    m_allocator->Free(m_counts);
 }
 
 // We need to lazy allocate the histogram data so static "histo" variables don't try to call the CLR memory allocator
```
Comment (adityamandaleeka) [inline src/jit/utils.cpp]: You may want to also update any comments like this that still refer to histo.

Comment (pgavlin) [inline src/jit/utils.cpp]: Done. Turns out it was just this one :)

**Comment by CarolEidt** [comment]
LGTM

**Pull request merged**